### DISCUSSION TITLE
Input for FAST_SFunc

### DISCUSSION THREAD

**Discussion by zhzenith** [OpenFAST/openfast#1163] [Q&A] [answered]
I found that the input for  S-function FAST_SFunc (v3.1.0) should be a one dimensional vector with 51 elements. Previously it was a one dimensional vector with 8 elements. Where can I find the description of the 51 elements? Thank you in advance for your kind help.

Best Regard,
zhzenith

**Comment by andrew-platt** [comment, marked as answer]
@zhzenith,

Version 3.0.0 only included 8 channels in the FAST_SFunc.  With the release of version 3.1.0, we added Simulink for 3 airfoil control channels, and 20 cable control channel groups (each with a commanded length change and rate of change channel).   The Simulink examples include these new sets of channels (https://github.com/OpenFAST/openfast/tree/v3.1.0/glue-codes/simulink/examples).  I realize in posting this reply that our documentation on this change is not obvious in the release notes, nor is it very well documented in the code itself -- we will add this info to the documentation to avoid future confusion.

**FAST_SFunc input channels for v 3.1.0**

1. Generator Torque (N-m)
2. Electrical Power (W)
3. Yaw pos (rad)
4. Yaw rate (rad/s)
5. Blade 1 pitch angle (rad)
6. Blade 2 pitch angle (rad)
7. Blade 3 pitch angle (rad)
8. High speed shaft brake fraction (-)
9. Blade 1 Airfoil control (-)
10. Blade 2 Airfoil control (-)
11. Blade 3 Airfoil control (-)
12. Cable control channel 1 DeltaL (m)
13. Cable control channel 2 DeltaL (m)
14. Cable control channel 3 DeltaL (m)
15. Cable control channel 4 DeltaL (m)
16. Cable control channel 5 DeltaL (m)
17. Cable control channel 6 DeltaL (m)
18. Cable control channel 7 DeltaL (m)
19. Cable control channel 8 DeltaL (m)
20. Cable control channel 9 DeltaL (m)
21. Cable control channel 10 DeltaL (m)
22. Cable control channel 11 DeltaL (m)
23. Cable control channel 12 DeltaL (m)
24. Cable control channel 13 DeltaL (m)
25. Cable control channel 14 DeltaL (m)
26. Cable control channel 15 DeltaL (m)
27. Cable control channel 16 DeltaL (m)
28. Cable control channel 17 DeltaL (m)
29. Cable control channel 18 DeltaL (m)
30. Cable control channel 19 DeltaL (m)
31. Cable control channel 20 DeltaL (m)
32. Cable control channel 1 DeltaLDot (m/s)
33. Cable control channel 2 DeltaLDot (m/s)
34. Cable control channel 3 DeltaLDot (m/s)
35. Cable control channel 4 DeltaLDot (m/s)
36. Cable control channel 5 DeltaLDot (m/s)
37. Cable control channel 6 DeltaLDot (m/s)
38. Cable control channel 7 DeltaLDot (m/s)
39. Cable control channel 8 DeltaLDot (m/s)
40. Cable control channel 9 DeltaLDot (m/s)
41. Cable control channel 10 DeltaLDot (m/s)
42. Cable control channel 11 DeltaLDot (m/s)
43. Cable control channel 12 DeltaLDot (m/s)
44. Cable control channel 13 DeltaLDot (m/s)
45. Cable control channel 14 DeltaLDot (m/s)
46. Cable control channel 15 DeltaLDot (m/s)
47. Cable control channel 16 DeltaLDot (m/s)
48. Cable control channel 17 DeltaLDot (m/s)
49. Cable control channel 18 DeltaLDot (m/s)
50. Cable control channel 19 DeltaLDot (m/s)
51. Cable control channel 20 DeltaLDot (m/s)

Cheers,
Andy

**Reply by zhzenith** [reply]
Well noted. Thank you @andrew-platt. If you have any information (paper) about "cable control", please share it with me. I really appreciate your kind help.

Best Regards,
zhzenith
